CMakeLists.txt

```diff
@@ -1,7 +1,10 @@
 project("PlaydateCPP")
 
+option(PDCPP_STAGE_IN_BINARY_DIR "Use CMake binary dir (instead of source dir) to stage files" OFF)
```

Comment:
I really like the motivation behind this change. It's always bothered me that the output of the pdc is in the project directory, but not enough to do anything about. I just end up adding `*.pdx` to all my `.gitignore`s and grumble about it. 

Anyway, the way this is implemented is blind to a user's other `Source`-type artifacts like any game-specific fonts, images, pdxinfo, audio sample files, etc. which need to come along for the ride before we can call this feature "done".

I'll leave it up to you how that gets implemented, maybe you have to specify where the input resources folder is or something, but either way I look forward to seeing what you come up with and merging it in.

Reply:
My expectation would be for the user to copy files via the `configure_file()` function in CMake (which also allows for content to be generated when CMake is run, e.g. to populate the `version` field of the `pdxinfo` file).

Here's an example:
```cmake
configure_file(
   "${RES_DIR}/Playdate/pdxinfo"
   "${PROJECT_BINARY_DIR}/Source/pdxinfo"
)
configure_file(
   "${RES_DIR}/Playdate/card.png"
   "${PROJECT_BINARY_DIR}/Source/card.png"
   COPYONLY
)
configure_file(
   "${RES_DIR}/Playdate/card-pressed.png"
   "${PROJECT_BINARY_DIR}/Source/card-pressed.png"
   COPYONLY
)
```

If you'd like, I could update the code to expose the `PDCPP_STAGING_DIR` variable to the `PARENT_SCOPE`.

Reply:
Gotcha, ok. My apologies if that was always in the PR description and I overlooked it.

I would love for this to be the default behaviour as a future iteration. To get there we'll need to have the ability for the user to have a "resources" folder where everything gets copied in automatically, (otherwise most devs with more than a few assets are going to prefer the current workflow,) but this is a worthy first step in that direction.